python/cppyy/numba_ext.py

```diff
@@ -130,14 +142,59 @@ def get_call_type(self, context, args, kwds):
         except KeyError:
             pass
 
-        ol = CppFunctionNumbaType(self._func.__overload__(*(numba2cpp(x) for x in args)), self._is_method)
+        args_mapping = []
+        for i, arg in enumerate(list(args)):
+            match = re.search(r"typeref\[(.*?)\*\]", str(arg))
+            if match:
+                literal_val = match.group(1)
+                arg_type = numba.typeof(eval(literal_val))
+                args_mapping.append(to_ref(numba2cpp(arg_type)))
+            else:
+                args_mapping.append(numba2cpp(arg))
+        args_combinations = itertools.product(*args_mapping)
+
+        for arg_combo in args_combinations:
+            try:
+                if(len(arg_combo) > 1):
+                    signature = ", ".join(arg_combo)
+                elif(len(arg_combo) == 0):
+                    signature = ""
+                else:
+                    signature = arg_combo[0]
```

Comment:
If I understand correctly, the error that we are trying to catch is not received through these statements, so these can be moved out of the try block.

Reply:
done!